File: apps/api/AssetManagement.Api/Models/Location.cs
namespace AssetManagement.Api.Models;

public class Location
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public string? Country { get; set; }
    public bool IsArchived { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Asset> Assets { get; set; } = [];
}

File: apps/web/src/types/location.ts
export interface Location {
  id: string;
  name: string;
  address: string | null;
  city: string | null;
  country: string | null;
  isArchived: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateLocationRequest {
  name: string;
  address?: string | null;
  city?: string | null;
  country?: string | null;
}

export interface UpdateLocationRequest {
  name: string;
  address?: string | null;
  city?: string | null;
  country?: string | null;
}

export interface LocationAsset {
  id: string;
  name: string;
  assetTypeName: string | null;
  status: string;
  assignedPersonName: string | null;
}

export interface LocationPerson {
  id: string;
  fullName: string;
  email: string | null;
  department: string | null;
  jobTitle: string | null;
}
